src/Common/nunit/TestSelectionParserException.cs

```diff
@@ -21,46 +21,41 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 // ***********************************************************************
 
-namespace NUnit.Framework.Internal.Filters
+using System;
+using System.Runtime.Serialization;
+
+#if NUNIT_ENGINE
+namespace NUnit.Engine
+#else
+namespace NUnit.Common
+#endif
 {
     /// <summary>
-    /// SimpleCategoryFilter parses a basic string representing a
-    /// single category or a list of categories separated by commas
+    /// TestSelectionParserException is thrown when an error 
+    /// is found while parsing the selection expression.
     /// </summary>
-    public class SimpleCategoryExpression
+#if !PORTABLE
+    [Serializable]
+#endif
+    public class TestSelectionParserException : Exception
     {
-        private string text;
-
-        private TestFilter filter;
-
         /// <summary>
-        /// Construct category filter from a text string
+        /// Construct with a message
         /// </summary>
-        /// <param name="text">A list of categories to parse</param>
-        public SimpleCategoryExpression(string text)
-        {
-            this.text = text;
-        }
+        public TestSelectionParserException(string message) : base(message) { }
 
         /// <summary>
-        /// Gets the TestFilter represented by the expression
+        /// Construct with a message and inner exception
         /// </summary>
-        public TestFilter Filter
-        {
-            get
-            {
-                if (filter == null)
```

Comment:
Yes, we could add that to our Compatibility classes. I'm personally in the
habit of just programming to the lowest version we support, so it doesn't
really bother me much. :-)

Percentage of 2.0 usage is quite low based on a survey we did, but it's a
decision we made to keep supporting it in NUnit 3.0. As a result, the
engine itself targets .NET 2.0. Of course, we will drop it sometime in the
future.

Remember, NUnit 2.6.4 still even supports .NET 1.1. I guess we are proud of
our backward compatibility. A test framework, in my view, has to be very
conservative about dropping support because old stuff still needs testing.

On Mon, Oct 12, 2015 at 2:21 AM, Igor Ziselman <notifications@github.com>
wrote:

> In src/Common/nunit/TestSelectionParserException.cs
> <https://github.com/nunit/nunit/pull/877#discussion_r41737459>:
>
> >          /// </summary>
> > -        public TestFilter Filter
> > -        {
> > -            get
> > -            {
> > -                if (filter == null)
>
> Custom version of Lazy can be implemented in .NET2.
> By the way, do you know percentage of nunit usage in .NET2 projects?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/877/files#r41737459>.
>